packages/rabbitmq/src/TxSubmitWorker.ts

```diff
@@ -109,13 +118,18 @@ export class TxSubmitWorker {
    * The common handler for errors
    *
    * @param isAsync flag to identify asynchronous errors
-   * @param err the error itself
+   * @param err the error itself supplied by 'close' handler runtime or on worker.start() catch block
    */
-  private async errorHandler(isAsync: boolean, err: unknown) {
+  private async errorHandler(isAsync: boolean, err?: any) {
     if (err) {
       this.logError(err, isAsync);
       this.#status = 'error';
       await this.stop();
+
+      // Emit event due to connection error in order to notify the worker abstraction listener
+      if (isConnectionError(err)) {
```

Comment:
I'm not so sure about this check.
We are not sure that `isConnectionError` includes all network related `Error`s that `amqplib` can throw and leave us prone to changes in the `amqplib` itself.
More than this `TxSubmitWorker.errorHandler` closes the worker regardless from the the error being a connection related one or not.
For the same reason I would have used the standard `'error'` event name.

Reply:
The thing here is that since we reuse [errorHander](https://github.com/input-output-hk/cardano-js-sdk/blob/39d22556809a465c07e6615c628707ff7ee19c29/packages/rabbitmq/src/TxSubmitWorker.ts#L116-L133) for both on startup and runtime error handling, I noticed that there could be passed as an `error` a `ProviderError` one as well, raised by [provider](https://github.com/input-output-hk/cardano-js-sdk/blob/39d22556809a465c07e6615c628707ff7ee19c29/packages/rabbitmq/src/TxSubmitWorker.ts#L165) not a healthy response. 
So my question here is requirements oriented, do we want to distinguish the errors (connection/non-connection) or to emit all types of errors from the worker error handler? 🤔

Reply:
Since the worker is closed anyway I would restart it anyway as well, otherwise we could have situations where the worker is stopped and not restarted: we will have in our hands a worker which is not running and which nobody will never restart.

Reply:
RIght, in that case where we always stop the worker caused by any type of error and put it into `idle` state, we should then handle the restarts via the worker abstraction with DNS resolution (regardless `ProviderError` error coming from Ogmios provider or errors coming from the `rabbitmq` broker ). Without service discovery - we entrust the restarts to nix/infrastructure we have.